### PR TITLE
[le11] chrome: update to 109.0.5414.74 and addon (1)

### PR DESCRIPTION
**Pull request by heitbaum**
- chrome: update 109.0.5414.74 and addon (1)

### CODE REVIEW

**Comment by CvH** [comment]
Does chrome extract? Wondering because @SupervisedThinking reported that the deb is not working for him

**Comment by heitbaum** [comment]
> Does chrome extract? Wondering because @SupervisedThinking reported that the deb is not working for him

I’ll boot both nuc6 and nuc12 into le10 and legacy respectfully in the morning and test. Too busy checking out the blood moon at the moment.

**Review by CvH — APPROVED** [review]

**Comment by CvH** [comment]
rebase pls

**Comment by SupervisedThinking** [comment]
@heitbaum did you test the tar / xz extraction?

**Comment by mglae** [comment]
@SupervisedThinking Can confirm extraction error. Recent Google chrome deb files do have two issues for us:

1. The ar archive contains illegal file mode 644 instead of 100644. Busybox ar is refusing this.
2. The xz/lzma compression is using CRC64. Busybox only implements CRC32.

@heitbaum Please include https://github.com/mglae/LibreELEC.tv/commits/le11_chrome_114_download

**Comment by SupervisedThinking** [comment]
@mglae how about
shipping binutils ar:
- https://github.com/SupervisedThinking/LibreELEC-RR/commit/241dd61733d45040317c22c54574d65a7f08e0f7
- https://github.com/SupervisedThinking/LibreELEC-RR/commit/e93db4d2435dad9ab4aad35691d4833c626af380

using xz binary instead binutils xz
- https://github.com/SupervisedThinking/LibreELEC-RR/commit/9286e50c08ef297de72f9b1867cde2ecd7634af3
- https://github.com/SupervisedThinking/LibreELEC-RR/commit/09cacf7d5b3a641339623e3b24ba6ddba68090fd

use xz & ar for the deb files
- https://github.com/SupervisedThinking/LibreELEC-RR/blob/master-rr/packages/supervisedthinking/web/brave-browser/scripts/brave-browser.start#L97-L99

= we don't need new packages & just use already build stuff

**Comment by CvH** [comment]
isn't z7ip supporting it too ?
also we can switch from legacy p7zip to 7zip these days too

**Comment by mglae** [comment]
@SupervisedThinking With backport in mind I decided to only change the addon and not the image. But if wanted the needed binaries can be added to the addon and called with absolute path.

**Comment by SupervisedThinking** [comment]
@mglae that's indeed a point... anyway mind to report your findings at https://bugs.busybox.net/ ?

**Comment by mglae** [comment]
@SupervisedThinking No. The behavior is bad for us but I don't see an error.

1. Busybox implements an archiver for the ar format whereas binutils ar "only" seem to be working with regular files in static libs.
2. xz behavior fulfil the [xz file format documentation](https://git.tukaani.org/?p=xz.git;a=blob;f=doc/xz-file-format.txt;h=4ed665060183375b8a1fb74384d01636042b1bcf;hb=24790f49ae66938c1c7574315e1c0aba1ed5ed25#l353). CRC64 is optional.

**Comment by SupervisedThinking** [comment]
I guess since https://github.com/LibreELEC/LibreELEC.tv/pull/7106 was merged we should check which xorg libs we need to rebuild

**Comment by heitbaum** [comment]
> I guess since #7106 was merged we should check which xorg libs we need to rebuild

I have rebased this. But next will be which chrome- libraries should be dropped

**Comment by SupervisedThinking** [comment]
IMHO all which where now build shared, we don't need to ship them for chrome once they are included anyway

**Comment by heitbaum** [comment]
Hi @mglae / @antonlacon - commit pulled in - and this PR "MERGE BLOCK" dropped.

**Comment by heitbaum** [comment]
> I believe this needs mglae's commits to replace the *.deb extraction with python from: https://github.com/mglae/LibreELEC.tv/commits/le11_chrome_114_download

Added. With a minor typographic fix in ad7e849945f21100302c87dbc5050f2a8a42eb0c

**Comment by mglae** [comment]
Working fine (read: chrome is starting) but being coauthor I cannot approve.

File: packages/addons/addon-depends/chrome-depends/unix_ar/package.mk [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# SPDX-License-Identifier: GPL-2.0
```
Comment (CvH) [inline packages/addons/addon-depends/chrome-depends/unix_ar/package.mk:1]:
```suggestion
# SPDX-License-Identifier: GPL-2.0-only
```

**Review by CvH — CHANGES_REQUESTED** [review]

File: packages/addons/addon-depends/chrome-depends/unix_ar/package.mk [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# Copyright (C) 2022-present Team LibreELEC (https://libreelec.tv)
+
+PKG_NAME="unix_ar"
+PKG_VERSION="0.2.1"
+PKG_SHA256="4e1c6e8fe5255b7babe93b9af51532ba09c6946eee413c5db00cab3878d7ed2f"
+PKG_LICENSE="BSD 3-Clause"
+PKG_SITE="https://github.com/getninjas/unix_ar"
+PKG_URL="https://github.com/getninjas/unix_ar/archive/refs/tags/${PKG_VERSION}.tar.gz"
+PKG_DEPENDS_TARGET=""
```
[the remark this answers is not in the document]
Comment (heitbaum) [reply packages/addons/addon-depends/chrome-depends/unix_ar/package.mk]: updated and squashed

File: packages/addons/addon-depends/chrome-depends/unix_ar/package.mk (hunk 1 above)
Comment (heitbaum) [reply packages/addons/addon-depends/chrome-depends/unix_ar/package.mk]: updated and squashed

File: packages/addons/browser/chrome/source/bin/deb_extract_data [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/python3
+
+# SPDX-License-Identifier: GPL-2.0
```
[the remark this answers is not in the document]
Comment (heitbaum) [reply packages/addons/browser/chrome/source/bin/deb_extract_data]: updated and squashed

**Review by CvH — APPROVED** [review]

**Pull request merged**